app/scripts/controllers/createFromURL.js

```diff
@@ -0,0 +1,205 @@
+    var createDetails = _.pick($routeParams, function(value, key) {
+      // routeParams without a value (e.g., ?app&) return true, which results in "true" displaying in the UI
+      return _.contains(whiteListedCreateDetailsKeys, key) && (value && value !== true);
+    });
+    // if no namespace is specified, set it to 'openshift'
+    createDetails.namespace = createDetails.namespace || 'openshift';
+
+    var getResources = function() {
+      if (createDetails.imageStream) {
+        DataService
+          .get("imagestreams", createDetails.imageStream, {namespace: createDetails.namespace}, {
+            errorNotification: false
+          })
+          .then(function(imageStream) {
+            DataService
+              .get("imagestreamtags", imageStream.metadata.name + ":" + createDetails.imageTag, {namespace: createDetails.namespace}, {
+                  errorNotification: false
+                })
+                .then(function(imageStreamTag){
+                  $scope.validationPassed = true;
+                  $scope.resource = imageStreamTag;
+                  createDetails.displayName = $filter('displayName')(imageStreamTag);
+                }, function(e){
+                  alertInvalidImageTag(createDetails.imageTag, e);
+                });
+          }, function(e) {
+            if(e.status === 403) {
+              alertInaccessibleNamespace(createDetails.namespace, e);
+            } else {
+              alertInvalidImageStream(createDetails.imageStream, e);
+            }
+          });
+      }
+      if (createDetails.template) {
+        DataService
+          .get("templates", createDetails.template, {namespace: createDetails.namespace}, {
+            errorNotification: false
+          })
+          .then(function(template) {
+            $scope.validationPassed = true;
+            $scope.resource = template;
+          }, function(e) {
+            if(e.status === 403) {
+              alertInaccessibleNamespace(createDetails.namespace, e);
+            } else {
+              alertInvalidTemplate(createDetails.template, e);
+            }
+          });
+      }
+    };
+
+    if (!(_.includes(namespaceWhitelist, createDetails.namespace))) {
+      alertInvalidNamespace(createDetails.namespace);
+    } else {
+      if (createDetails.imageStream && createDetails.template) {
+        showInvalidResource();
+      } else if (!(createDetails.imageStream) && !(createDetails.template)) {
+        alertResourceRequired();
+      } else {
+        getResources();
+      }
+    }
+
+    angular.extend($scope, {
+      createDetails: createDetails,
+      createWithProject: function(project) {
+        $location
+          .url('/project/' + (project ||  $scope.selected.project.metadata.name) + '/create/' + ($routeParams.imageStream ? 'fromimage' : 'fromtemplate'))
+          .search(createDetails);
```

Comment:
We have this URL built in a few places. Probably want to consolidate it in Navigate service (and have filters call the service when needed). See

https://github.com/spadgett/origin-web-console/blob/2c15578e8963210ce91e795ff515f9c845883940/app/scripts/services/navigate.js#L72-L85

https://github.com/spadgett/origin-web-console/blob/2c15578e8963210ce91e795ff515f9c845883940/app/scripts/filters/resources.js#L619-L644